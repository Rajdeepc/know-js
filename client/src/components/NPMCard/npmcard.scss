.npm-card {
    h5 {
        font-weight: bold;
    }

    p.card-text {
        display: block;
        display: -webkit-box;
        -webkit-line-clamp: 3;
        -moz-line-clamp: 3;
        -ms-line-clamp: 3;
        -o-line-clamp: 3;
        line-clamp: 3;
        -webkit-box-orient: vertical;
        -ms-box-orient: vertical;
        -o-box-orient: vertical;
        box-orient: vertical;
        overflow: hidden;
        text-overflow: ellipsis;
        white-space: normal;
        height: 56px;
        font-size: 12px;
        margin-bottom: 10px;
    }

    .npm-tags {
        overflow: auto;

        .nav-item {
            float: left;
            background: #7d7d7d;
            color: #fff;
            margin-bottom: 10px;
            margin-right: 6px;

            a {
                color: #fff;
                padding: 4px 10px;
                font-size: 10px;
            }
        }
    }
}