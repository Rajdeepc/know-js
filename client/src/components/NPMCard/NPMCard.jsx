import React from "react";
import { Card, Nav } from "react-bootstrap";
import "./npmcard.scss";
import moment from "moment";
import PropTypes from "prop-types";

const NPMCard = ({ item }) => {
  return (
    <div className="npm-card" data-testid="test-npmcard-card-wrapper">
      <Card>
        <a
          target="_blank"
          rel="noopener noreferrer"
          href={item.package && item.package.links && item.package.links.npm}
        >
          <Card.Body>
            <h5>{item.package.name}</h5>
            <Card.Text>{item.package.description}</Card.Text>
            <div className="npm-tags">
              {(item.package.keywords || [])
                .slice(0, 4)
                .map((itemKey, index) => {
                  return (
                    <Nav.Item key={`index-${index}`}>
                      <Nav.Link
                        href={
                          item.package &&
                          item.package.links &&
                          item.package.links.npm
                        }
                      >
                        {itemKey}
                      </Nav.Link>
                    </Nav.Item>
                  );
                })}
            </div>
            <p>
              Author:{" "}
              {item.package && item.package.author && item.package.author.name}
            </p>
            <p>
              Last Updated:{" "}
              {item.package.date &&
                moment(item.package.date).format("MM/DD/YYYY")}
            </p>
          </Card.Body>
        </a>
      </Card>
    </div>
  );
};

NPMCard.defaultProps = {
  package: {
    name: "",
    description: "",
    author: {
      name: "",
    },
    date: "",
    keywords: [],
    links: {
      npm: "",
    },
  },
};

NPMCard.propTypes = {
  package: PropTypes.objectOf(PropTypes.shape({
    name: PropTypes.string,
    description: PropTypes.string,
    author: PropTypes.shape({
      name: PropTypes.string,
    }),
    date: PropTypes.string,
    keywords: PropTypes.arrayOf,
    links: PropTypes.shape({
      npm: PropTypes.string,
    }),
  }))
};
export default NPMCard;
