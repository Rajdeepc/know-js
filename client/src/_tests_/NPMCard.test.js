import React from "react";
import { render, cleanup } from "@testing-library/react";
import NPMCard from "../components/NPMCard/NPMCard";
import "@testing-library/jest-dom/extend-expect";

afterEach(cleanup);

const props = {
      package: {
        name: "",
        description: "",
        author: {
          name: "",
        },
        date: "",
        keywords: [],
        links: {
          npm: "",
        },
      }
};

describe("<NPMCard component>", () => {
  test("should be defined", () => {
    expect(NPMCard).toBeDefined();
  });
  test("renders NPMCard Component", () => {
    const { container, rerender } = render(<NPMCard item={props} />);
    rerender(<NPMCard item={props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
  test("renders for default Props", () => {
      const { getByTestId } = render(
          <NPMCard item={props}
          />
      );
      expect(getByTestId('test-npmcard-card-wrapper')).toBeDefined();
      expect(NPMCard.defaultProps.package.name).toBeDefined()
  });
  //open_issues returns correctly
  test('package object returns correctly' , () => {
      const result = NPMCard.defaultProps.package.name;
      expect(result).toBe('')
  })
});
