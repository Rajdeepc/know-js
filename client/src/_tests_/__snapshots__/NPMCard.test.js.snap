// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NPMCard component> renders NPMCard Component 1`] = `
<div
  class="npm-card"
  data-testid="test-npmcard-card-wrapper"
>
  <div
    class="card"
  >
    <a
      href=""
      rel="noopener noreferrer"
      target="_blank"
    >
      <div
        class="card-body"
      >
        <h5 />
        <p
          class="card-text"
        />
        <div
          class="npm-tags"
        />
        <p>
          Author:
           
          
        </p>
        <p>
          Last Updated:
           
          
        </p>
      </div>
    </a>
  </div>
</div>
`;
